### DISCUSSION TITLE
[Question] Connecting the Library with Another Library

### DISCUSSION THREAD

**Discussion by devanshiAppperfect** [battery-data-alliance/pymacnet#19] [Q&A]
**Question -**
I am interested in integrating this library with another library in my project. Could you please provide guidance on how to achieve this connection? Are there any recommended practices or considerations to keep in mind?

**My requirements -**
1. I want to integrate this library to another battery data stimulating library
2. I want to use the interface's methods in my library 

**Any specific steps or code examples would be greatly appreciated.**

Thank you

**Comment by chintanp** [comment]
@devanshiAppperfect Please be specific about what you want to try to achieve. pymacnet is just a wrapper around MacNet. If you can think of a use for MacNet, then pymacnet maybe able to help you achieve that using Python. Also to be noted, at present, not all MacNet functionality is possible through pymacnet.

**Comment by devanshiAppperfect** [comment]
@chintanp I am creating my own library to plot battery data and I want to connect it with pymacnet to get that battery data.
Also I would like to use it with another battery data stimulating library to get the data from there and use it for pymacnet.

**Comment by chintanp** [comment]
@devanshiAppperfect we provide examples of how to use pymacnet. 

What do you mean by library ? Can you give examples of exactly what you want to try to achieve? What have you tried so far ? What Maccor version do you have ? 

What programming language are you using ? How will you plot the data, in Python with Matplotlib etc. Or in Grafana etc?

**Comment by devanshiAppperfect** [comment]
@chintanp We’re trying to get data out of pymacnet. We haven't tried much as we are not able to create a link. The programming language that we are using for integration is python3.11 and the plotting would be done using plotly dash and also we might use Grafana.

**Comment by chintanp** [comment]
@devanshiAppperfect thank you for providing the info. Could you please describe what you mean by "not able to create a link" in your post above? Also, which Maccor cycler and software version on it do you have? PyMacNet receives data from a [Maccor Battery Cycler](http://www.maccor.com/), so that one to be provisioned by you and the IP address of the same updated in the config file.
